cmd/nfd-topology-updater/main.go

```diff
@@ -0,0 +1,135 @@
+
+	if *printVersion {
+		fmt.Println(ProgramName, version.Get())
+		os.Exit(0)
+	}
+
+	// Assert that the version is known
+	if version.Undefined() {
+		klog.Warningf("version not set! Set -ldflags \"-X sigs.k8s.io/node-feature-discovery/pkg/version.version=`git describe --tags --dirty --always`\" during build or run.")
+	}
+
+	// Plug klog into grpc logging infrastructure
+	utils.ConfigureGrpcKlog()
+
+	klConfig, err := kubeconf.GetKubeletConfigFromLocalFile(resourcemonitorArgs.KubeletConfigFile)
+	if err != nil {
+		klog.Fatalf("error getting topology Manager Policy: %v", err)
+	}
+	tmPolicy := string(topologypolicy.DetectTopologyPolicy(klConfig.TopologyManagerPolicy, klConfig.TopologyManagerScope))
+	klog.Infof("Detected kubelet Topology Manager policy %q", tmPolicy)
+
+	// Get new TopologyUpdater instance
+	instance, err := topology.NewTopologyUpdater(args, resourcemonitorArgs, tmPolicy)
+	if err != nil {
+		klog.Exitf("failed to initialize TopologyUpdater instance: %v", err)
+	}
+
+	if err = instance.Run(); err != nil {
+		klog.Exit(err)
+	}
+}
+
+func parseArgs(flags *flag.FlagSet, osArgs ...string) (*topology.Args, *resourcemonitor.Args) {
+	args, resourcemonitorArgs := initFlags(flags)
+
+	_ = flags.Parse(osArgs)
+	if len(flags.Args()) > 0 {
+		fmt.Fprintf(flags.Output(), "unknown command line argument: %s\n", flags.Args()[0])
+		flags.Usage()
+		os.Exit(2)
+	}
+
+	return args, resourcemonitorArgs
+}
+
+func initFlags(flagset *flag.FlagSet) (*topology.Args, *resourcemonitor.Args) {
+	args := &topology.Args{}
+	resourcemonitorArgs := &resourcemonitor.Args{}
+
+	flagset.StringVar(&args.CaFile, "ca-file", "",
+		"Root certificate for verifying connections")
+	flagset.StringVar(&args.CertFile, "cert-file", "",
+		"Certificate used for authenticating connections")
+	flagset.StringVar(&args.KeyFile, "key-file", "",
+		"Private key matching -cert-file")
+	flagset.BoolVar(&args.Oneshot, "oneshot", false,
+		"Update once and exit")
+	flagset.BoolVar(&args.NoPublish, "no-publish", false,
+		"Do not publish discovered features to the cluster-local Kubernetes API server.")
+	flagset.StringVar(&args.KubeConfigFile, "kube-config-file", "",
+		"Kube config file.")
+	flagset.DurationVar(&resourcemonitorArgs.SleepInterval, "sleep-interval", time.Duration(60)*time.Second,
+		"Time to sleep between CR updates. Non- positive value implies no re-labeling (i.e. infinite sleep). [Default: 60s]")
+	flagset.StringVar(&resourcemonitorArgs.Namespace, "watch-namespace", "",
+		"Namespace to watch pods. Use \"\" for all namespaces.")
+	flagset.StringVar(&resourcemonitorArgs.KubeletConfigFile, "kubelet-config-file", "/podresources/config.yaml",
+		"Kubelet config file path.")
+	flagset.StringVar(&resourcemonitorArgs.PodResourceSocketPath, "podresources-socket", "/podresources/kubelet.sock",
+		"Pod Resource Socket path to use.")
+	flagset.StringVar(&resourcemonitorArgs.FsRoot, "fsroot", "/host",
+		"Mount point of the filesystem root.")
```

Comment:
Hmm, I think we should/could just use `source.pathPrefix` and drop this command line flag. If we need to have `/host/` then adjust for this elsewhere. I think it would make sense to have `/host/boot`, `/host/sys/` etc. instead of `/host-boot`, `/host-sys`...

Reply:
Sure, using source.pathPrefix and dropped the command line flag.

Reply:
This seems to have creeped back when squashing 😄  -> drop